internal/sdk/trace.go

```diff
@@ -0,0 +1,153 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sdk
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"runtime"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/embedded"
+)
+
+func GetTracerProvider() trace.TracerProvider {
+	return tracerProviderInstance
+}
+
+var tracerProviderInstance TracerProvider
+
+type TracerProvider struct {
+	embedded.TracerProvider
+}
+
+var _ trace.TracerProvider = TracerProvider{}
+
+func (p TracerProvider) Tracer(name string, opts ...trace.TracerOption) trace.Tracer {
+	cfg := trace.NewTracerConfig(opts...)
+	return Tracer{
+		name:      name,
+		version:   cfg.InstrumentationVersion(),
+		schemaURL: cfg.SchemaURL(),
+	}
+}
+
+type Tracer struct {
+	embedded.Tracer
+
+	name, schemaURL, version string
+}
+
+var _ trace.Tracer = Tracer{}
+
+func (t Tracer) Start(ctx context.Context, name string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
+	cfg := trace.NewSpanStartConfig(opts...)
+	s := Span{
+		tracer:      t,
+		name:        name,
+		spanContext: trace.SpanContextFromContext(ctx),
+		attributes:  cfg.Attributes(),
+		links:       cfg.Links(),
+		newRoot:     cfg.NewRoot(),
+		spanKind:    cfg.SpanKind(),
+		stackTrace:  cfg.StackTrace(),
+	}
+	if t := cfg.Timestamp(); !t.IsZero() {
+		s.startTime = t.UnixNano()
+	}
+	return t.embed(ctx, s)
+}
+
+func (t Tracer) embed(ctx context.Context, s trace.Span) (context.Context, trace.Span) {
+	return trace.ContextWithSpan(ctx, s), s
+}
+
+type Span struct {
+	embedded.Span
+
+	tracer      Tracer
+	name        string
+	spanContext trace.SpanContext
+	attributes  []attribute.KeyValue
+	startTime   int64
+	links       []trace.Link
+	newRoot     bool
+	spanKind    trace.SpanKind
+	stackTrace  bool
+}
+
+func (s Span) SpanContext() trace.SpanContext {
+	return s.spanContext
+}
+
+func (s Span) IsRecording() bool {
```

Comment:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#recording-sampled-reaction-table

This needs to return a value set during `embed` that the eBPF func will set a field. We need to return true here if sampled, and false otherwise. We do not plan to send the span to a processor if it is not sampled.